### DISCUSSION TITLE
Playing with Docker + io.containerd.spin.v1

### DISCUSSION THREAD

**Discussion by GordonSmith** [spinframework/spin#1718] [Q&A] [answered]
I have been able to `spin new` and host in docker the following languages:
* c
* python
* rust (but had to downgrade the SDK to v1.3.0 and wit-bindgen to rev "cb871cfa1ee460b51eb1d144b175b9aab9c50aba")

But for ts + js I am getting the following error:
```
docker: Error response from daemon: Others("could not build spin trigger: Failed to instantiate component 'spin-docker-ts'"): unknown.
```

Any thoughts?

**Comment by itowlson** [comment, marked as answer]
Kia ora @GordonSmith, which version of the TS/JS plugin/SDK are you using?  (`spin plugins list --installed`)  As you've seen with Rust, the latest `spn.v1` containerd shim has not yet been upgraded to handle the Spin 1.4 bindings.  If you're on the latest SDK, could you please try this:

* Downgrade the `js2wasm` plugin to version 0.4.0 (`spin plugin upgrade js2wasm -v 0.4.0 --downgrade`)
* Change the `@fermyon/spin-sdk` reference in `package.json` to `"^0.4.0"`

Then try rebuilding and see if that helps?  Thanks!

**Reply by GordonSmith** [reply]
Thx!

```
cloud 0.1.1 [installed]
js2wasm 0.5.0 [installed]
py2wasm 0.3.0 [installed]
```

I will downgrade and report back!

**Reply by GordonSmith** [reply]
Yup - that fixed it!

**Reply by itowlson** [reply]
Great, thanks for closing the loop on that!
